### ISSUE TITLE
How to use l2_regularizer right in a custom policy?

### ISSUE THREAD

**Issue by zkx741481546** [hill-a/stable-baselines#839]
**Describe the bug**
I'm using kernel_regularizer = l2_regularizer(l2_rate) in my dense layer of my custom policy, but the training reward curve is exactly same when I use different l2_rate. Do I need to change  the loss function manually? Or there are any other ways to make this right?

**Code example**
```python
import gym
import tensorflow as tf
from stable_baselines.common.policies import ActorCriticPolicy, register_policy, nature_cnn, MlpPolicy, \
    FeedForwardPolicy
from stable_baselines.common.vec_env import DummyVecEnv
from stable_baselines import A2C
from Config import seed

class CustomPolicy(ActorCriticPolicy):
    def __init__(self, sess, ob_space, ac_space, n_env, n_steps, n_batch, reuse=False, **kwargs):
        super(CustomPolicy, self).__init__(sess, ob_space, ac_space, n_env, n_steps, n_batch, reuse=reuse, scale=True)

        with tf.variable_scope("model", reuse=reuse):
            activ = tf.nn.relu
            net_arch = [dict(vf=[256, 128, 64, 32], pi=[256, 128, 64, 32])]
            l2_scale = 0.01
            dropout_rate = 0.0
            training = False
            for k, v in kwargs.items():
                if k == 'act_fun':
                    activ = v
                elif k == 'net_arch':
                    net_arch = v
                elif k == 'l2_scale':
                    l2_scale = v
                elif k == 'dropout_rate':
                    dropout_rate = v
                    if dropout_rate == 0.:
                        training = False
                    else:
                        training = True
                else:
                    raise Exception("不支持的Policy_arg:{}".format(k))
            # extracted_features = nature_cnn(self.processed_obs, **kwargs)
            extracted_features = self.processed_obs
            if len(ob_space.shape) > 1:
                extracted_features = tf.layers.flatten(extracted_features)
            index_code = 0
            arch_pi_and_vf = None
            for arch in net_arch:
                if isinstance(arch, int):
                    extracted_features = tf.layers.dropout(activ(
                        tf.layers.dense(extracted_features, arch, name='feature_extract' + str(arch),
                                        kernel_regularizer=tf.contrib.layers.l2_regularizer(l2_scale))),
                        rate=dropout_rate,
                        seed=seed,
                        training=training)
                    index_code += 1
                elif isinstance(arch, dict):
                    arch_pi_and_vf = arch
                else:
                    raise Exception("自定义网络参数不合法: {}".format(arch))
            pi_layer_size = arch_pi_and_vf['pi']
            vf_layer_size = arch_pi_and_vf['vf']
            pi_h = extracted_features
            for i, layer_size in enumerate(pi_layer_size):
                pi_h = tf.layers.dropout(activ(tf.layers.dense(pi_h, layer_size, name='pi_fc' + str(i),
                                                               kernel_regularizer=tf.contrib.layers.l2_regularizer(
                                                                   l2_scale))), rate=dropout_rate, seed=seed,
                                         training=training)
            pi_latent = pi_h

            vf_h = extracted_features
            for i, layer_size in enumerate(vf_layer_size):
                vf_h = tf.layers.dropout(activ(tf.layers.dense(vf_h, layer_size, name='vf_fc' + str(i),
                                                               kernel_regularizer=tf.contrib.layers.l2_regularizer(
                                                                   l2_scale))), rate=dropout_rate, seed=seed,
                                         training=training)
            value_fn = tf.layers.dense(vf_h, 1, name='vf')
            vf_latent = vf_h

            self._proba_distribution, self._policy, self.q_value = \
                self.pdtype.proba_distribution_from_latent(pi_latent, vf_latent, init_scale=0.01)

        self._value_fn = value_fn
        self._setup_init()

    def step(self, obs, state=None, mask=None, deterministic=False):
        if deterministic:
            action, value, neglogp = self.sess.run([self.deterministic_action, self.value_flat, self.neglogp],
                                                   {self.obs_ph: obs})
        else:
            action, value, neglogp = self.sess.run([self.action, self.value_flat, self.neglogp],
                                                   {self.obs_ph: obs})
        return action, value, self.initial_state, neglogp

    def proba_step(self, obs, state=None, mask=None):
        return self.sess.run(self.policy_proba, {self.obs_ph: obs})

    def value(self, obs, state=None, mask=None):
        return self.sess.run(self.value_flat, {self.obs_ph: obs})


```

And there is a sample of my policy_kwargs:
`policy_args = dict(act_fun=gelu, net_arch=[dict(vf=[256, 128], pi=[256, 128])], l2_scale=0.01,
                   dropout_rate=0.)`

**Comment by Miffyli** [comment]
You need to update the loss too (see [this old-ish example](https://github.com/cscheau/Examples/blob/master/iris_l1_l2.py), which still probably holds). This means you have to delve into the algorithm, find where the loss is computed, and add those regularizations there.

Note for v3: Should probably allow custom policies to modify the loss(es) or provide an additional one, to make this kind of regularizations easier.
